src/dosext/misc/emm.c

```diff
@@ -273,33 +273,44 @@ static void set_map_registers(const struct emm_reg *buf, int pages);
 #define Kdebug1(args)		E_Stub args
 #define Kdebug2(args)		E_Stub args
 
+static int ems_init_done = 0;
+
 void
 ems_helper(void) {
   switch (LWORD(ebx)) {
   case 0:
     E_printf("EMS Init called!\n");
     if (!config.ems_size) {
+      LWORD(eax) = 1;	/* report error */
```

Comment:
It's to fit with ems.S doing this at line 235
```ASM
orb %al, %al
jz 1f
```
Before this patch it's only set on success, which means on error it has whatever value was left by the call. But I'm happy to change, i just worry about going off and changing loads of unrelated stuff unnecessarily.